src/util/asmap.cpp

```diff
@@ -77,9 +77,12 @@ uint32_t Interpret(const std::vector<bool> &asmap, const std::vector<bool> &ip)
             return DecodeASN(pos, endpos);
         } else if (opcode == 1) {
             jump = DecodeJump(pos, endpos);
+            if (pos + jump < pos) {
+              break; // bail on overflow
+            }
             if (bits == 0) break;
             if (ip[ip.size() - bits]) {
-                if (jump >= endpos - pos) break;
+                if (pos + jump >= endpos) break;
                 pos += jump;
             }
```

Comment:
Shouldn't the added check for overflow be inside `if (ip[ip.size() - bits])`? If we don't get inside that `if` then we don't care whether it overflows because we would not do the arithmetic that would overflow.

nit: only 2 spaces used for indentation, the rest of the code uses 4.

Reply:
Fixed, thanks.